components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultPatientsResourceImplTest.java

```diff
@@ -0,0 +1,199 @@
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.phenotips.data.PatientRepository;
+import org.phenotips.data.rest.DomainObjectFactory;
+import org.phenotips.data.rest.PatientsResource;
+import org.phenotips.data.rest.model.Patient;
+import org.phenotips.data.rest.model.Patients;
+import org.xwiki.component.manager.ComponentLookupException;
+import org.xwiki.component.manager.ComponentManager;
+import org.xwiki.component.util.DefaultParameterizedType;
+import org.xwiki.component.util.ReflectionUtils;
+import org.xwiki.context.Execution;
+import org.xwiki.context.ExecutionContext;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.model.reference.EntityReferenceResolver;
+import org.xwiki.query.Query;
+import org.xwiki.query.QueryException;
+import org.xwiki.query.QueryManager;
+import org.xwiki.query.internal.DefaultQuery;
+import org.xwiki.rest.XWikiRestException;
+import org.xwiki.security.authorization.AuthorizationManager;
+import org.xwiki.security.authorization.Right;
+import org.xwiki.test.mockito.MockitoComponentMockingRule;
+import org.xwiki.users.User;
+import org.xwiki.users.UserManager;
+
+import javax.inject.Provider;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriInfo;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.slf4j.Logger;
+
+public class DefaultPatientsResourceImplTest {
+
+    @Rule
+    public MockitoComponentMockingRule<PatientsResource> mocker =
+        new MockitoComponentMockingRule<PatientsResource>(DefaultPatientsResourceImpl.class);
+
+    @Mock
+    private User currentUser;
+
+    @Mock
+    private Logger logger;
+
+    private PatientRepository repository;
+
+    private QueryManager queries;
+
+    private AuthorizationManager access;
+
+    private UserManager users;
+
+    private EntityReferenceResolver<EntityReference> currentResolver;
+
+    private DomainObjectFactory factory;
+
+    @Mock
+    private Patient patient;
+
+    private URI uri;
+
+    @Mock
+    private UriInfo uriInfo;
+
+    private DefaultPatientsResourceImpl patientsResource;
+
+    private XWikiContext context;
+
+
+    @Before
+    public void setUp() throws ComponentLookupException, URISyntaxException {
+        MockitoAnnotations.initMocks(this);
+        Execution execution = mock(Execution.class);
+        ExecutionContext executionContext = mock(ExecutionContext.class);
+        ComponentManager compManager = this.mocker.getInstance(ComponentManager.class, "context");
+        when(compManager.getInstance(Execution.class)).thenReturn(execution);
+        doReturn(executionContext).when(execution).getContext();
+        doReturn(mock(XWikiContext.class)).when(executionContext).getProperty("xwikicontext");
```

Comment:
Instead of returning a mock XWikiContext return the XWikiContext created by the `MockitoComponentMockingRule ` this also makes use of the `context` field you have written but are not currently using
here is how you do that:
```java
Provider<XWikiContext> provider = this.mocker.getInstance(XWikiContext.TYPE_PROVIDER);
this.context = provider.get();
...
doReturn(this.context).when(executionContext).getProperty("xwikicontext");
```